sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuiteBase.scala

```diff
@@ -1249,7 +1257,12 @@ abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
           val e = intercept[IllegalArgumentException] {
             cast(Literal.create(interval), dataType).eval()
           }.getMessage
-          assert(e.contains("Interval string does not match day-time format"))
+          assert(e.contains(s"Interval string does not match day-time format of " +
+            s"${IntervalUtils.supportedFormat((dataType.startField, dataType.endField))
+              .map(format => s"`$format`").mkString(", ")} " +
+            s"when cast to ${dataType.typeName}: $interval, " +
+            s"set spark.sql.legacy.fromDayTimeString.enabled to true " +
```

Comment:
Let's use `LEGACY_FROM_DAYTIME_STRING.key`.

Reply:
> Let's use `LEGACY_FROM_DAYTIME_STRING.key`.

Done and tested in local